### PR TITLE
Change extension formalism, add docs about it

### PR DESCRIPTION
**Pull request by Lnaden**
Changed how extending a simulation works by reducing syntax length and removing ambiguity with `number_of_iterations` in the YAML file. Added documentation on the `running.rst` page to help users access this feature.

### CODE REVIEW

File: Yank/repex.py [hunk 1]
```diff
@@ -924,7 +923,7 @@ def _initialize_create(self):
         """
 
         if self._initialized:
-            raise Error("Simulation has already been initialized.")
+            raise Exception("Simulation has already been initialized.")
```
Comment (andrrizzi) [inline Yank/repex.py]: Maybe `RuntimeError` instead of general `Exception`?

File: docs/running.rst [hunk 2]
```diff
@@ -170,3 +170,43 @@ For example, to force YANK to use the ``OpenCL`` platform:
...
+
+First you set :ref:`yaml_options_number_of_iterations` to an integer number of iterations you wish to extend the
+simulation. If no simulation has been run yet, then one will be run for the number of iterations.
+Setting :ref:`yaml_options_extension_simulation` to ``True`` modifies the behavior of
+:ref:`yaml_options_number_of_iterations` to extend the simulation by the specified number, adding on to what is already
+on the file.
+
+One could optionally just increase :ref:`yaml_options_number_of_iterations`, but then you have to change
+the YAML file every time you want to extend the run. Setting :ref:`yaml_options_extension_simulation` allows you to run
+the same YAML file without modification to do the same thing.
```
Comment (andrrizzi) [inline docs/running.rst]: It's probably worth pointing out that if your combinatorial job times out and you resume with the extension flag, it's going to extend also the simulations that have been already completed.

**Comment by jchodera** [comment]
Can we also document how to extend simulations via the command-line?

**Comment by Lnaden** [comment]
Any other changes before I merge this?

**Comment by andrrizzi** [comment]
Looks good to me!

**Pull request merged**